tests/TestCase/Http/ServerRequestTest.php

```diff
@@ -1228,6 +1228,7 @@ public function testGetDataOnStringData(): void
         $request = new ServerRequest(compact('post'));
         $this->assertNull($request->getData('Model'));
         $this->assertNull($request->getData('Model.field'));
+        $this->assertSame($post, $request->getParsedBody());
```

Comment:
Return value of getParsedBody() can't be a string as per the interface https://github.com/php-fig/http-message/blob/efd67d1dc14a7ef4fc4e518e7dee91c271d524e4/src/ServerRequestInterface.php#L164

Reply:
I think we should just throw an exception if the constructor receives as string value as parsed body. We already have the `BodyParserMiddleware` which handles setting the parsed body as array/object from the string body contents https://github.com/cakephp/cakephp/blob/bff920a50e0dd0912e24915a778b0913f57225b2/src/Http/Middleware/BodyParserMiddleware.php#L168.

Reply:
Throwing an error if the parsed results are strings makes sense.

Reply:
I have pushed a commit to that effect.

Reply:
This looks good. Can you change the test function description (and name?) just for consistency?

Reply:
n/m, I can do this.